power_grid_model_c/power_grid_model/include/power_grid_model/main_model.hpp

```diff
@@ -273,12 +274,13 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         is_asym_parameter_up_to_date_ = is_asym_parameter_up_to_date_ && !changes.topo && !changes.param;
     }
 
-    template <math_output_type MathOutputType, typename MathSolverType, typename InputType, typename PrepareInputFn,
-              typename SolveFn>
+    template <math_output_type MathOutputType, typename MathSolverType, typename YBus, typename InputType,
+              typename PrepareInputFn, typename SolveFn>
         requires std::invocable<std::remove_cvref_t<PrepareInputFn>> &&
-                 std::invocable<std::remove_cvref_t<SolveFn>, MathSolverType&, InputType const&> &&
+                 std::invocable<std::remove_cvref_t<SolveFn>, MathSolverType&, InputType const&, YBus const&> &&
```

Comment:
here and below, can you change the order into `MathSolverType&, YBus const&, InputType const&`?